### DISCUSSION TITLE
How combine line plot and mesh plot?

### DISCUSSION THREAD

**Discussion by Sauradeep92** [bwoodsend/vtkplotlib#18] [Q&A] [answered]
I am trying to combine two plots in a single plot window. Basically, the line plot will appear inside the bounds of the mesh, if plotted together correctly. Here is the relevant snippet:

```
import vtkplotlib as vpl
from stl.mesh import mesh

for i in axons:
 vpl.plot(i, color ="dark red")

# Read the STL using numpy-stl
mesh = Mesh.from_file(path)

# Plot the mesh
vpl.mesh_plot(mesh)

# Show the figure
vpl.show()
```
I keep getting only the second mesh plot without the first line plot. To get both, I have to plot them separately in two windows. However, that's not what I want.

This module has no attribute "hold" so unable to use something of that sort too. Any help is greatly appreciated.

Thank you.

**Comment by bwoodsend** [comment, marked as answer]
Are you sure that your axons aren't just much much smaller than your mesh or possibly inside your mesh? That would make them look like they've disappeared when they haven't.

Everything you plot goes into the same figure - it's not reset until you call `vpl.show()` or `vpl.figure()` - so it should be doing exactly what you want it to do. For example if I add definitions to your `axons` and `path` variables so that I can run your example...

```python
import vtkplotlib as vpl
from stl.mesh import Mesh
import numpy as np

axons = np.random.uniform(-100, 100, (3, 5, 3))
for i in axons:
 vpl.plot(i, color ="dark red")

# Read the STL using numpy-stl
path = vpl.data.get_rabbit_stl()
mesh = Mesh.from_file(path)

# Plot the mesh
vpl.mesh_plot(mesh)

# Show the figure
vpl.show()
```
... gives me the image below:

![image](https://user-images.githubusercontent.com/30940778/189479974-675560ad-21d0-4f4e-a9f5-e1190cfa59e9.png)

**Reply by Sauradeep92** [reply]
You are correct in assuming that the axons should be inside the mesh and smaller, but they should still be visible. I am not exactly sure what's going on. I am unable to attach my data because stl or npy file types are not supported.

**Reply by bwoodsend** [reply]
If you put your data in a zip file, GitHub will let you upload that.

**Reply by Sauradeep92** [reply]
Thanks. I have attached the two inputs: array and mesh, which I am trying to plot. The data itself should be fine because I can plot them individually, so my best guess is this is probably a minor visualization issue? I appreciate any comments/suggestions you have.
[Input.zip](https://github.com/bwoodsend/vtkplotlib/files/9541204/Input.zip)

**Reply by bwoodsend** [reply]
Not only are the axons tiny but they're also miles away from your mesh. Your axons are centered around the origin the mesh's Z values range from 146.7499 to 153.25.

```python
>>> mesh.min_
array([ -1.5   ,  -1.5   , 146.7499], dtype=float32)
>>> mesh.max_
array([  1.5 ,   1.5 , 153.25], dtype=float32)
>>> axons.min((0, 1))
array([-0.00033039, -0.00016   ,  0.0125    ])
>>> axons.max((0, 1))
array([0.00033039, 0.000521  , 0.2875    ])
```
If I move the mesh so that it's centered around the origin (use `mesh.vectors[:, :, 2] -= mesh.z.mean()` before the `vpl.mesh_plot(...)` call), then zoom into the middle of the mesh, I can see the lines.

![image](https://user-images.githubusercontent.com/30940778/189498891-4ac6af06-7b6f-41e0-8df6-806ebf56a3c8.png)

**Reply by Sauradeep92** [reply]
Thank you for your time and pointing this out, it was of great help, I really appreciate it. I messed up with the units. 
![image](https://user-images.githubusercontent.com/22203681/189510470-7096a7c1-80a3-401d-be8c-d06a5904f94f.png)

**Reply by bwoodsend** [reply]
You're welcome. [You're not the first person to screw up units](https://github.com/bwoodsend/vtkplotlib/discussions/6).
